packages/frontend/src/components/common/Filters/FilterInputs/DateFilterInputs.tsx

```diff
@@ -236,6 +239,45 @@ const DateFilterInputs: FC<FilterInputsProps<DateFilterRule>> = (props) => {
                     />
                 </MultipleInputsWrapper>
             );
+        case FilterOperator.IN_BETWEEN:
+            return (
+                <StyledDateRangeInput
+                    className={disabled ? 'disabled-filter' : ''}
+                    disabled={disabled}
+                    formatDate={(value: Date) =>
+                        formatDate(value, undefined, false)
+                    }
+                    parseDate={parseDate}
+                    value={
+                        filterRule.values
+                            ? [
+                                  new Date(filterRule.values?.[0]),
+                                  new Date(filterRule.values?.[1]),
```

Comment:
When the user changes from one operator to `in between` the `filterRule.values` is an array with a single value. 
We could have instead something like this:
```
value= {[
  filterRule.values?.[0] ? new Date(filterRule.values?.[0]) : null,
  filterRule.values?.[1] ? new Date(filterRule.values?.[1]) : null
]}
```
https://user-images.githubusercontent.com/9117144/211606044-dd8992cd-20ef-4468-bb5a-449ecfc574e4.mp4

Note that the input style doesn't seem to be working like the other inputs. Usually, this means the blueprint input component needs a `fill` property.

Reply:
good catch!
had to override blueprint's css for this component to work in the popover filter tab on dashboard, so this is an expected side effect
Priyanka said to keep it this way for now since we're changing libraries soon